### PR TITLE
flamenco: fixing loader state read leaks

### PR DESCRIPTION
**Pull request by ibhatt-jumptrading**
https://github.com/firedancer-io/auditor-internal/issues/83
https://github.com/firedancer-io/auditor-internal/issues/96

### CODE REVIEW

File: src/flamenco/runtime/program/fd_bpf_loader_v3_program.c [hunk 1]
```diff
@@ -50,7 +50,7 @@ fd_bpf_loader_v3_is_executable( fd_exec_slot_ctx_t * slot_ctx,
   fd_bincode_decode_ctx_t ctx = {
     .data    = (uchar *)meta     + meta->hlen,
     .dataend = (char *) ctx.data + meta->dlen,
-    .valloc  = slot_ctx->valloc,
+    .valloc  = fd_scratch_virtual(),
```
Comment (topointon-jump) [inline src/flamenco/runtime/program/fd_bpf_loader_v3_program.c:53]:
```suggestion
    .valloc  = instr_ctx->valloc,
```

File: src/flamenco/runtime/program/fd_bpf_loader_v3_program.c [hunk 2]
```diff
@@ -94,7 +90,7 @@ read_bpf_upgradeable_loader_state_for_program( fd_exec_txn_ctx_t *
   fd_bincode_decode_ctx_t ctx = {
     .data    = rec->const_data,
     .dataend = rec->const_data + rec->const_meta->dlen,
-    .valloc  = txn_ctx->valloc,
+    .valloc  = fd_scratch_virtual(),
```
Comment (topointon-jump) [inline src/flamenco/runtime/program/fd_bpf_loader_v3_program.c:93]:
```suggestion
    .valloc  = instr_ctx->valloc,
```

**Review by topointon-jump — APPROVED** [review]

File: src/flamenco/runtime/program/fd_bpf_loader_v3_program.c (hunk 1 above)
Comment (ibhatt-jumptrading) [reply src/flamenco/runtime/program/fd_bpf_loader_v3_program.c:53]: we don't have access to the instr context in this function

File: src/flamenco/runtime/program/fd_bpf_loader_v3_program.c (hunk 2 above)
Comment (ibhatt-jumptrading) [reply src/flamenco/runtime/program/fd_bpf_loader_v3_program.c:93]: same as above

**Pull request merged**